test/unit/instrumentation/undici.test.js

```diff
@@ -136,6 +145,51 @@ tap.test('undici instrumentation', { skip: shouldSkip }, function (t) {
         t.end()
       })
     })
+
+    t.test(
+      'should get the parent segment executionAsyncResource when it already exists',
```

Comment:
I was thinking of adding a unit test with 2 diff async resources but I have integration tests that will verify this